test.c

```diff
@@ -2267,6 +2267,35 @@ test_refresh_timeout ()
     CU_ASSERT (assert_apteryx_empty ());
 }
 
+
+void
+test_refresh_duplicate ()
+{
+    const char *path = TEST_PATH"/interfaces/eth0/state";
+    const char *path2 = TEST_PATH"/interfaces/eth0/*";
+    char *value = NULL;
+
+    _cb_count = 0;
+    _cb_timeout = TEST_SLEEP_TIMEOUT / 2;
+    CU_ASSERT (apteryx_refresh (path, test_refresh_callback));
+    CU_ASSERT (apteryx_refresh (path2, test_refresh_callback));
```

Comment:
What is the behaviour we are trying to prevent here? If you register 2 refreshers should not both be called if the data is our of date? Or are we saying that if anyone updates the data then no refreshers are called. Perhaps.

Reply:
Yeah, the latter. I think it's a configuration issue to have two processes / callbacks trying to keep the same information updated. If that does happen I'd prefer that only one actually updates it (rather than re-writing the same section twice one after the other).

Reply:
yup - agreed - would be nice to have the test check that the behaviour is deterministic - in that only the first registered refresher is called.... or perhaps it should be the first registered that is closet to the trunk of the tree